src/main.rs

```diff
@@ -267,17 +268,77 @@ fn build(config: &Config) {
     };
 }
 
+fn static_file_handler(dest: &str, req: Request, res: Response) {
+    let path = PathBuf::from(format!("{}{}", dest, req.uri));
+
+    let serve_path = if path.is_file() {
+        // try to point the serve path to `path` if it corresponds to a file
+        path.clone()
+    } else {
+        // try to point the serve path into a "index.html" file in the requested
+        // path
+        path.join("index.html")
+    };
+
+    // if the request points to a file and it exists, read and serve it
+    if serve_path.exists() {
+        let mut file = match File::open(serve_path) {
+            Ok(f) => f,
+            Err(e) => {
+                error!("{}", e);
+                std::process::exit(1);
+            }
+        };
+
+        let mut buffer: Vec<u8> = vec![];
+
+        match file.read_to_end(&mut buffer) {
+            Err(e) => {
+                error!("{}", e);
+                std::process::exit(1);
+            },
+            Ok(_) => {},
+        };
+
+        match res.send(&buffer) {
+            Err(e) => {
+                error!("{}", e);
+                std::process::exit(1);
+            },
+            Ok(()) => {},
+        };
+    }
```

Comment:
You need to handle the `else` case here. I believe it should return a `404` error when it can not find the file.